backend/app/main.py

```diff
@@ -103,25 +196,72 @@ async def validation_exception_handler(request, exc):
     responses={'500': {'model': ErrorResponse}},
 )
 async def get_search(
-    query: Optional[str] = Query(None, description="query"), #optional for the sake of pagination
+    query: Optional[str] = Query(None, description="query"), # Used in for full text search
+    condition: Optional[str] = Query(None, description="listing condition"), # Used for sorting by condition
     limit: Optional[int] = Query(5, description="Number of listings to retrieve", ge=1, le=30), 
-    next: Optional[str] = Query(None, description="Last seen pagination token")
+    next: Optional[str] = Query(None, description="Last seen pagination token"),
+    type: str = Query(None, description="Type of search")
     ) -> Union[SearchGetResponse, ErrorResponse]:
     """
     search listings
     """
+    print("TYPE: ", type)
+    print("CONDITION: ", condition)
+    limit = min(max(limit, 1), 30)
     try:
-        limit = min(max(limit, 1), 30)
-        try:
-            listings = await get_listings(query=query, limit=limit, next=next)
-       
-            return SearchGetResponse(listings=listings.listings, 
+        # Funneling the type of search
+        if type == "full-text":
+            try:
+                listings = await get_listings(query=query, limit=limit, next=next)
+            except ErrorResponse as e:
+                        return e
+        
+        if type == "condition":
+            try:
+                cursor = listings_collection.find({"condition": condition})
+                listings = await cursor.to_list(length=limit) 
+            except ErrorResponse as e:
```

Comment:
The idea here is to:
1. get the document matching a full-text search
2. filter those documents by the condition
3. order them either by price or by date (not both)

Our discussion on discord was about this step 3 where we decided that we can't combine both price and date ordering. However, the first 2 steps only concern filtering which should always be possible regardless of what type of ordering we use for the results.